test/unit/chain/stateTransition/block/operations/index.test.ts

```diff
@@ -0,0 +1,223 @@
+import {expect} from "chai";
+import sinon from "sinon";
+import {hashTreeRoot} from "@chainsafe/ssz";
+
+import {Crosslink} from "../../../../../../src/types";
+import  * as processProposerSlashing
+  from "../../../../../../src/chain/stateTransition/block/operations/proposerSlashing";
+import  * as processAttesterSlashing
+  from "../../../../../../src/chain/stateTransition/block/operations/attesterSlashing";
+import  * as processAttestation
+  from "../../../../../../src/chain/stateTransition/block/operations/attestation";
+import  * as processDeposit
+  from "../../../../../../src/chain/stateTransition/block/operations/deposit";
+import  * as processVoluntaryExit
+  from "../../../../../../src/chain/stateTransition/block/operations/voluntaryExit";
+import  * as processTransfer
+  from "../../../../../../src/chain/stateTransition/block/operations/transfer";
+
+import {processOperations} from "../../../../../../src/chain/stateTransition/block/operations";
+import {generateState} from "../../../../../utils/state";
+import {generateEmptyBlock} from "../../../../../utils/block";
+import {generateDeposit} from "../../../../../utils/deposit";
+import {generateEmptyTransfer} from "../../../../../utils/transfer";
+import {
+  MAX_ATTESTATIONS,
+  MAX_ATTESTER_SLASHINGS,
+  MAX_DEPOSITS,
+  MAX_PROPOSER_SLASHINGS, MAX_VOLUNTARY_EXITS
+} from "../../../../../../src/constants";
+import {generateEmptyAttesterSlashing, generateEmptyProposerSlashing} from "../../../../../utils/slashings";
+import {generateAttestationData, generateEmptyAttestation} from "../../../../../utils/attestation";
+import {generateEmptyVoluntaryExit} from "../../../../../utils/voluntaryExits";
+import {MAX_TRANSFERS} from "../../../../../../src/constants/minimal";
+
+
+describe('process block - process operations', function () {
+
+  const sandbox = sinon.createSandbox();
+
+  let processProposerSlashingStub,
+    processAttesterSlashingStub,
+    processAttestationStub,
+    processDepositStub,
+    processVoluntaryExitStub,
+    processTransferStub;
+
+  beforeEach(() => {
+    processProposerSlashingStub = sandbox.stub(processProposerSlashing, "processProposerSlashing");
+    processAttesterSlashingStub = sandbox.stub(processAttesterSlashing, "processAttesterSlashing");
+    processAttestationStub = sandbox.stub(processAttestation, "processAttestation");
+    processDepositStub = sandbox.stub(processDeposit, "processDeposit");
+    processVoluntaryExitStub = sandbox.stub(processVoluntaryExit, "processVoluntaryExit");
+    processTransferStub = sandbox.stub(processTransfer, "processTransfer");
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('should fail to process operations - outstanding deposits not processed up to the maximum', function () {
+    const state  = generateState();
+    const body = generateEmptyBlock().body;
+    body.deposits.push(generateDeposit());
+    try {
+      processOperations(state, body);
```

Comment:
Maybe use expect(...).to.throw() here and elsewhere
eg:
```typescript
expect(() => processOperations(state, body)).to.throw();
```

Reply:
ok